### ISSUE TITLE
Is there such a thing as "derivation path" on TON?

### ISSUE THREAD

**Issue by akegaviar** [akegaviar/ton-foo#378]
In the Bitcoin standard there is concept called "derivation path" defined in BIP32: it is used by hierarchical deterministic wallets to derive keys.

How do TON wallets generate keys from the mnemonic phrase, do they have some analogue of derivation path?

This question was imported from Telegram Chat: [https://t.me/tondev/125625](https://t.me/tondev/125625)

## Answer

Good question! Professionals in the industry often discuss wallets in the context of `BIP32` (also known as Bitcoin Improvement Proposal), which underlies the process of how wallets are generated from mnemonics.

To put it simply, you can refer to this TypeScript file to understand how we can generate a wallet address:

[https://github.com/toncenter/tonweb-mnemonic/blob/master/src/functions/mnemonic-to-seed.ts](https://github.com/toncenter/tonweb-mnemonic/blob/master/src/functions/mnemonic-to-seed.ts)

As you can see, `pbkdf2Sha512` appears to be a combination of PBKDF2 (Password-Based Key Derivation Function 2) and Sha512, both of which are cryptographic methods.

---

[Original](https://answers.ton.org/question/1574159353718509568/is-there-such-a-thing-as-derivation-path-on-ton)
